src/subprocess-posix.cc

```diff
@@ -29,6 +29,18 @@ extern char** environ;
 
 #include "util.h"
 
+#define HANDLE_EAGAIN(x) ({ \
+  __typeof__(x) eintr_wrapper_result; \
+  do { \
+    eintr_wrapper_result = (x); \
+    if (eintr_wrapper_result == -1 && errno == EAGAIN) \
+      sleep(1); \
```

Comment:
Instead of sleeping, is there any practical work that Ninja might do somewhere else before revisiting the call to posix_spawn ?

Reply:
When this condition occurs it is generally a kernel-level race condition for locks. It may be possible to continue to do more work in user-space, but this fix is intended to prevent the ninja process from simply giving up and exiting, which it currently does. If possible, I'd like to avoid trying to solve both of those problems at the same time.